src/toil/leader.py

```diff
@@ -1107,10 +1107,11 @@ def innerLoop(jobStore, config, batchSystem, toilState, jobBatcher, serviceManag
                     # Remove the job
                     if jobGraph.remainingRetryCount > 0:
                         jobBatcher.issueJob(JobNode.fromJobGraph(jobGraph))
-                        logger.debug("Job: %s is empty, we are scheduling to clean it up", jobGraph.jobStoreID)
+                        logger.debug("Job: %s is empty, we are scheduling to clean it up", jobGraph)
                     else:
                         jobBatcher.processTotallyFailedJob(jobGraph)
-                        logger.warn("Job: %s is empty but completely failed - something is very wrong", jobGraph.jobStoreID)
+                        logger.critical("Job: %s is empty but completely failed - something is "
```

Comment:
Talked to @benedictpaten offline, he said an exception should be raised here & that if we do see that exception then it's indicative of a bug in the job store clean code.